src/lib.rs

```diff
@@ -438,10 +441,10 @@ impl VectorSelector {
 
         VectorSelector {
             first_bucket_max_time: start_time,
-            last_bucket_max_time: last_bucket_max_time,
-            end_time: end_time,
-            bucket_width: bucket_width,
-            lookback: lookback,
+            last_bucket_max_time,
+            end_time,
+            bucket_width,
+            lookback,
             elements: vec![None; num_buckets as usize],
```

Comment:
Disclamer: I don't know much about Rust style but should these be grouped by style? Named and unnamed parameters.